import-scripts/clear-persistence-caches-for-changed-studies.sh

```diff
@@ -0,0 +1,450 @@
+        selected_cache_reset_work_task_db="$portal_db"
+        tmp_study_subdir="$TMP_DIR_ROOT/$portal_db"
+        for filepath in $tmp_study_subdir/*.$STUDY_CACHE_INVALIDATION_NEEDED_SUFFIX ; do
+            if [ -f "$filepath" ] ; then
+                if flock_available_for_filepath "$filepath" ; then
+                    selected_cache_reset_work_task_trigger_filepath="$filepath"
+                    return 0
+                fi
+            fi
+        done
+    done
+    # nothing is pending
+    return 1
+}
+
+# examines database tmp subdir for presence of a matching succeeded status file, given a database label and a trigger filepath
+# usage : trigger_filepath_is_marked_as_succeeded db_label trigger_filepath
+function trigger_filepath_is_marked_as_succeeded() {
+    db="$1"
+    trigger_filepath="$2"
+    tmp_study_subdir="$TMP_DIR_ROOT/$db"
+    filename="$(basename -- $trigger_filepath)"
+    filerootname="${filename%.*}"
+    trigger_filepath="$tmp_study_subdir/$filerootname.$STUDY_CACHE_INVALIDATION_NEEDED_SUFFIX"
+    succeeded_filepath="$tmp_study_subdir/$filerootname.$STUDY_CACHE_INVALIDATION_SUCCEEDED_SUFFIX"
+    if [ -f "$succeeded_filepath" ] ; then
+        return 0 # yes, marked as succeeded
+    fi
+    return 1
+}
+
+# clears the persistence cache, either using the cache api for study cache reset tasks, or the kubectl based script for full database cache clearing tasks
+# proper function depends on having the selected cache reset work task set in global environment variables
+# usage : perform_selected_cache_reset_task
+function perform_selected_cache_reset_task() {
+    # if (unlikely) task has been successfully performed while we selected it, exit
+    if trigger_filepath_is_marked_as_succeeded "$selected_cache_reset_work_task_db" "$selected_cache_reset_work_task_trigger_filepath" ; then
+        return 0 # someone else completed he work already
+    fi
+    # we are already in a subshell (from the caller) and hold the lock for selected_cache_reset_work_task_trigger_filepath
+    # handle full cache reset if triggered
+    trigger_filename="$(basename -- $selected_cache_reset_work_task_trigger_filepath)"
+    if [ "$trigger_filename" == "$CACHE_RESET_NEEDED_FILENAME" ] ; then
+        /data/portal-cron/scripts/clear_cbioportal_persistence_cache.sh "$selected_cache_reset_work_task_db"
+        return $?
+    fi
+    # handle study cache reset if triggered and able
+    if ! database_can_clear_per_study_cache "$selected_cache_reset_work_task_db" ; then
+        return 1 # not able - fail
+    fi
+    # trigger is per-study cache reset (by deduction)
+    study_id="${trigger_filename%.*}"
+    cache_api_key="${portal_db_to_cache_api_key[$selected_cache_reset_work_task_db]}"
+    cache_api_base_uri="${PORTAL_DB_TO_PORTAL_API_BASE_URI[$selected_cache_reset_work_task_db]}"
+    curl -X DELETE -H "X-API-KEY:$cache_api_key" -v -i "$cache_api_baseuri/" "$cache_api_base_uri/cache/$study_id"
```

Comment:
Double check why two baseuri's are here.

Reply:
that was an error. Thanks. That was fixed, and I removed the "-v" (verbose) option and the "-i" (include response headers) option to help reduce logging noise. I've also sent the standard output to /dev/null.

Reply:
Actually, I tested and curl returns status 0 (success) even when the endpoint returns a failure message. Instead of redirecting to null, I pipe to grep to see if a "success" word is present in output. The word "Flushed" will appear in the output on success, so I test for it.